src/ui/layout/mct-tree.vue

```diff
@@ -57,43 +99,275 @@
+
+            if(currentLocationPath && !alreadyNavigated) {
+                this.scrollTo = descendantPath.pop();
+                descendantPath = descendantPath.join('/');
+                this.allChildren = [];
+                this.ancestors = [];
+                this.jumpPath = descendantPath;
+                this.jumpToPath(true);
+            } else if(alreadyNavigated && !this.noScroll) {
+                this.scrollTo = descendantPath.pop();
+                this.autoScroll();
+            }
         }
     },
     mounted() {
+        let savedPath = this.getSavedNavigatedPath();
+        if(savedPath) {
+            this.jumpPath = savedPath;
+            this.afterJump = () => {
+                if(this.currentPathIsActivePath()) {
+                    this.scrollTo = this.currentlyViewedObjectId();
+                }
+            }
+        }
         this.searchService = this.openmct.$injector.get('searchService');
-        this.getAllChildren();
+        window.addEventListener('resize',  this.handleWindowResize);
+        this.openmct.objects.get('ROOT').then(root => {
+            let rootNode = this.buildTreeItem(root);
+            this.getAllChildren(rootNode);
+        });
+    },
+    destroyed() {
+        window.removeEventListener('resize', this.handleWindowResize);
     },
     methods: {
-        getAllChildren() {
+        updatevisibleTreeItems() {
+            if (!this.updatingView) {
+                this.updatingView = true;
+                requestAnimationFrame(()=> {
+                    let start = 0,
+                        end = this.pageThreshold,
+                        allItemsCount = this.allTreeItems.length;
+
+                    if (allItemsCount < this.pageThreshold) {
+                        end = allItemsCount;
+                    } else {
+                        let firstVisible = this.calculateFirstVisibleItem(),
+                            lastVisible = this.calculateLastVisibleItem(),
+                            totalVisible = lastVisible - firstVisible,
+                            numberOffscreen = this.pageThreshold - totalVisible;
+
+                        start = firstVisible - Math.floor(numberOffscreen / 2);
+                        end = lastVisible + Math.ceil(numberOffscreen / 2);
+
+                        if (start < 0) {
+                            start = 0;
+                            end = Math.min(this.pageThreshold, allItemsCount);
+                        } else if (end >= allItemsCount) {
+                            end = allItemsCount;
+                            start = end - this.pageThreshold + 1;
+                        }
+                    }
+                    this.itemOffset = start;
+                    this.visibleTreeItems = this.allTreeItems.slice(start, end);
+
+                    this.updatingView = false;
+                });
+            }
+        },
+        setContainerHeight() {
+            if(this.$refs.mainTree) {
```

Comment:
Using vue.$nextTick should be a more predictable way to know when the tree is rendered

Reply:
Yep, changed to nextTick and that covered the replacement!